spec/unit/http_client_spec.js

```diff
@@ -226,6 +226,82 @@ describe('HttpClient', function () {
       });
     });
 
+    it('serializes multipart/related request correctly', function () {
+      let request = {
+        verb: 'POST',
+        path: '/',
+        headers: {
+          'Content-Type': 'multipart/related'
+        },
+        partHeaders: {
+          input_field_name: {
```

Comment:
Is there a requirement for individual parts to have their own set of headers? If so, is there any way we can encapsulate them into the part itself? 

I'm thinking about something like: 

```js
let req = {
  verb: 'POST',
  path: '/',
  headers: {},
  body: {
    file_part: fs.createReadStream('...'),
    form_part: 'some-value',
    other_form_part: new FormPart({
      whatever_key: 'some-value',
      headers: {
        'Content-Type': 'application/json'
      }
    })
  }
};
```

Reply:
I suppose this applies mostly for the content type of a given object in a body, but the point still stands.